src/com/codefortomorrow/advanced/chapter16/solutions/LinkedList.java

```diff
@@ -0,0 +1,153 @@
+package com.codefortomorrow.advanced.chapter16.solutions;
+
+// UUID to represent each node's unique ID
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+// using Lombok for Getters and Setters (saves lines)
+import lombok.Setter;
+import lombok.Setter;
+
+/**
+ * @author ArmeetJatyani
+ * March 2021
+ *
+ * Implement a simple LinkedList
+ * You will need to create a LinkedListNode class, which represents each item/node in the linked list
+ * Required functionality...
+ * - head(): get head of list
+ * - tail(): get tail of list
+ * - add(): add to tail of list
+ * - push(): push to head of list
+ * - pop(): remove head of list
+ * - toString(): return a String representation of the list
+ */
+
+@Setter
+public class LinkedList {
+
+    private LinkedListNode head;
+
+    /**
+     * default constructor
+     */
+    public LinkedList() {
+        head = null;
+    }
+
+    /**
+     * constructor with first value
+     * @param value
+     */
+    public LinkedList(int value) {
+        // create first node
+        head = new LinkedListNode(value, null);
+    }
+
+    /**
+     * get head of Linked List
+     * @return
+     */
+    public LinkedListNode head() {
+        return this.head;
+    }
+
+    /**
+     * traverse and get tail of linked list
+     * @return
+     */
+    public LinkedListNode tail() {
+        LinkedListNode current = head;
+        if (current == null) return null;
+
+        while (current.getNext() != null) {
+            current = current.getNext();
```

Comment:
```suggestion
            current = current.next();
```

Reply:
Maybe you're using the getter created by the `lombok` module? But I noticed the `LinkedListNode` also has a `next()` method so I figured why not use that one.